library/src/main/java/com/kizitonwose/calendarview/ui/DayHolder.kt

```diff
@@ -21,28 +24,21 @@ internal data class DayConfig(
 internal class DayHolder(private val config: DayConfig) {
 
     private lateinit var dateView: View
-    private lateinit var containerView: FrameLayout
     private lateinit var viewContainer: ViewContainer
-
-    var day: CalendarDay? = null
+    private var day: CalendarDay? = null
 
     fun inflateDayView(parent: LinearLayout): View {
         dateView = parent.inflate(config.dayViewRes).apply {
-            // We ensure the layout params of the supplied child view is
-            // MATCH_PARENT so it fills the parent container.
-            layoutParams = layoutParams.apply {
-                height = ViewGroup.LayoutParams.MATCH_PARENT
-                width = ViewGroup.LayoutParams.MATCH_PARENT
-            }
-        }
-        containerView = FrameLayout(parent.context).apply {
             // This will be placed in the WeekLayout(A LinearLayout) hence we
             // use LinearLayout.LayoutParams and set the weight appropriately.
             // The parent's wightSum is already set to 7 to accommodate seven week days.
-            layoutParams = LinearLayout.LayoutParams(config.width, config.height, 1F)
-            addView(dateView)
+            updateLayoutParams<LinearLayout.LayoutParams> {
+                width = config.width - getMarginStart(this) - getMarginEnd(this)
```

Comment:
Why do you subtract the margins here? The calendar will multiply the `config.width` value by 7 to get the size for each month's width. So, wouldn't it mean that subtracting the margins will give a wrong value?